RandomObjects.py

```diff
@@ -91,16 +92,47 @@ def generateSeq(self):
         """Calculates a random size and location to generate a randomized shape
         then calls recursiveGen() many times to generate the shape
         """
-        sizeScalarW = int(math.sqrt(self.height) * 1.2)
+        sizeScalarW = int(math.sqrt(self.width) * 1.2)
         sizeScalarH = int(math.sqrt(self.height) * 1.2)
         sizeScalar = min(sizeScalarH, sizeScalarW)
         sizeScalar = random.randint(int(sizeScalar / 4), sizeScalar)
+        # robot's right most initial x position
+        robot_right = self.width / 2 + robot_radius / tile_size + 1
+        # robot's left most initial x position
+        robot_left = self.width / 2 - robot_radius / tile_size - 1
+        # robot's down most initial y position
+        robot_down = self.height / 2 - robot_radius / tile_size - 1
+        # robot's up most initial y position
+        robot_up = self.height / 2 + robot_radius / tile_size + 1
         goodLoc = False
         while not goodLoc:
             randX = random.randint(0, self.width - sizeScalar)
             randY = random.randint(0, self.height - sizeScalar)
-            if (
-                    randY + sizeScalar >= self.height or randX + sizeScalar >= self.width or randY - sizeScalar <= 0 or randX - sizeScalar <= 0):
+            # object's left most x position
+            object_left = randX - sizeScalar - (robot_radius / tile_size) * bloat_factor
+            # object's right most x position
+            object_right = randX + sizeScalar + (robot_radius / tile_size) * bloat_factor
+            # object's up most y position
+            object_up = randY + sizeScalar + (robot_radius / tile_size) * bloat_factor
+            # object's down most y position
+            object_down = randY - sizeScalar - (robot_radius / tile_size) * bloat_factor
```

Comment:
Great job including the bloat factor into this. 

Just as a sanity check make sure that ((robot_radius) / tile_size) * bloat_factor is the right units ie tiles